library/Icinga/Application/ApplicationBootstrap.php

```diff
@@ -605,7 +605,7 @@ protected function setupErrorHandling()
         ini_set('display_startup_errors', 1);
         ini_set('display_errors', 1);
         set_error_handler(function ($errno, $errstr, $errfile, $errline) {
-            if (error_reporting() === 0) {
+            if (!(error_reporting() & $errno)) {
```

Comment:
I'm not sure whether the bitwise *and* is correct here.

Previously, the condition was true if `error_reporting()` reported exactly `0`. Now it is true if the integer return value `error_reporting()` and the integer value of `$errno` have no common bits (i.e. no bits at the same position set).

Reply:
Don't worry, it's correct. The former check was wrong and has led to a bunch of misleading exceptions with PHP 8. Please see the [set_error_handler()](https://www.php.net/manual/en/function.set-error-handler.php) documentation for a related example.

The error is now ignored in case the configured error reporting flags (it's a bitmask, `0x7FFF` in our case) do not match the raised error level. `$errno` is 0 in case you're using the STFU-operator (`@`). And that's what this change fixes.